src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -538,6 +538,16 @@ private static void WriteDataSources(DirectoryWriter dir, CanvasDocument app, Er
                     }
                 }
 
+                foreach (var ds in dataSourceStateToWrite.Where(ds => ds.Type == "ViewInfo"))
+                {
+                    if (dataSourceDef != null && ds.Name.StartsWith(dataSourceDef.DatasetName))
```

Comment:
nit: can we encapsulate this in its own function? gives us a place to also heavily comment what's going on here.